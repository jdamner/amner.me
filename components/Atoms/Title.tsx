import React from "react";

import { font } from "utils/header-font";

export default function Title({ children }: React.PropsWithChildren) {
  const classes = [
    "tracking-tight",
    "uppercase",
    "text-6xl",
    "md:text-7xl",
    "lg:text-8xl",
    "text-orange-100",
    "mb-5",
    font.className,
  ];

  return <h2 className={classes.join(" ")}>{children}</h2>;
}
