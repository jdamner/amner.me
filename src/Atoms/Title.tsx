import React from "react";

import TwoThirds from "../Layouts/TwoThirds";
import ExpandingLine from "./ExpandingLine";

export default function Title({ children, title }: { children?: React.ReactNode, title?: string }) {

    const classes = [
        'text-lg',
        'font-sans',
        'font-semibold',
        'tracking-tight',
        'text-left',
        'font-slate-800',
        'uppercase',
        'text-5xl',
        'md:text-6xl',
        'break-words',
        'basis-1/3',
    ];

    const ChildComponent = <div className='w-full text-left  text-2xl font-black'>
        {children}
        <ExpandingLine />
    </div>


    return (
        title ?
            <TwoThirds first={<h2 className={classes.join(' ')}>{title}</h2>}>
                {children && ChildComponent}
            </TwoThirds> :
            children && ChildComponent
    );
}
