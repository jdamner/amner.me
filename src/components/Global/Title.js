import React from "react";

import Seperator from "./Seperator";
import { motion } from "framer-motion";

export default function Title({ text }) {
    return (
        <motion.div
            className='blog-wrap-title'
            initial={{ opacity: 0, y: -100 }}
            viewport={{ once: true}}
            whileInView={{ opacity: 1, y: 0 }}
            transition={{ duration: 0.5 }}
        >
            <Seperator />
            <h2>{text}</h2>
            <Seperator modifier='rotate' />
        </motion.div>
    );
}