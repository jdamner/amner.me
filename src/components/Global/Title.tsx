/* API */
import { motion, useReducedMotion } from "framer-motion";

/* Types */
import type { WithChildren } from "../../types/WithChildren.type";
import TwoThirds from "../Layouts/TwoThirds";

/**
 * Title Component
 * 
 * @param {children, string} props
 * @returns {JSX.Element}
 */
export default function Title({ children, title }: WithChildren & { title?: string }): JSX.Element {

    const classes = [
        'text-lg',
        'font-sans',
        'font-semibold',
        'tracking-tight',
        'text-left',
        'font-black',
        'uppercase',
        'text-xl',
        'md:text-6xl',
        'break-words',
        'basis-1/3',
    ];

    const reducedMotion = useReducedMotion();

    return (
        <TwoThirds first={title && <h2 className={classes.join(' ')}>{title}</h2>}>
            {children && <>
                <div className='w-full text-left uppercase text-2xl font-black'>
                    <div className='mb-2'>{children}</div>
                    <motion.div
                        className='bg-black h-2 w-full dark:bg-slate-300'
                        initial={{ maxWidth: reducedMotion ? '100%' : '0%' }}
                        viewport={{ once: true }}
                        whileInView={{ maxWidth: '100%' }}
                        transition={{ duration: 0.5 }}
                    >
                    </motion.div>
                </div>
            </>
            }
        </TwoThirds>
    );
}
