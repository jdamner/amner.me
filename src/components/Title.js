import React from "react";
import Corner from "./Global/Corner";


export default function About({ image, title, text }) {
    return (
        <section id="title" className="title">
            <div className="title-wrap">
                <div className='title-image'>
                    { image ?? null }
                </div>

                <div className="title-content">
                    <Corner rotation={-90} />
                    <h1 className='title-content-title'>{ title }</h1>
                    { text ?? null }
                </div>
            </div>
        </section>
    );
}