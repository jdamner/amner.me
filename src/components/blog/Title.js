import React from "react";
import Corner from "../global/Corner";


export default function About({ image, title, text }) {
    return (
        <section id="about" className="about">
            <div className="about-wrap">
                <div className='about-image'>
                    { image ?? null }
                </div>

                <div className="about-content">
                    <Corner rotation={-90} />
                    <h1 className='about-content-title'>{ title }</h1>
                    { text ?? null }
                </div>
            </div>
        </section>
    );
}