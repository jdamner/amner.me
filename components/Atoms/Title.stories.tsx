import type { Meta, StoryObj } from "@storybook/nextjs";
import { faker } from "@faker-js/faker";

import Title from "./Title";

type Story = StoryObj<typeof Title>;

export default {
  title: "Atoms/Title",
  component: Title,
} as Meta;

faker.seed(123);

export const Default: Story = {
  args: {
    children: faker.animal.dog(),
  },
};
