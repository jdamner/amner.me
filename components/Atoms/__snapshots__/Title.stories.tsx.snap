// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Title Default smoke-test 1`] = `
<div class="bg-black text-orange-100  font-sans">
  <h2 class="tracking-tight uppercase text-6xl md:text-7xl lg:text-8xl text-orange-100 mb-5 bebas-neue-normal-400">
    Podenco Canario
  </h2>
</div>
`;
